packages/flutter_tools/lib/src/xcode_project.dart

```diff
@@ -149,8 +149,13 @@ abstract class XcodeBasedProject extends FlutterProjectPlatform {
   /// Checks if FlutterGeneratedPluginSwiftPackage has been added to the
   /// project's build settings by checking the contents of the pbxproj.
   bool get flutterPluginSwiftPackageInProjectSettings {
```

Comment:
The new tests for skipping migrated PBXNativeTargets was failing at first. It turns out that this getter is a bit conservative, so it would also abort the migrator when the `pbxproj` would for example have `FlutterGeneratedPluginSwiftPackage` in the local swift package section even if the active target didn't have the migration. I updated all the relevant getters that were affected.

The new implementation now checks for a line in the `PBXBuildFile` section.